### PR TITLE
Add mining pools distribution metric

### PR DESCRIPTION
**Pull request by kalkov**
#### Summary

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex [hunk 1]
```diff
@@ -30,6 +31,21 @@ defmodule SanbaseWeb.Graphql.Resolvers.ClickhouseResolver do
     end
   end
 
+  def mining_pools_distribution(
+        _root,
+        %{from: from, to: to, interval: interval},
+        _resolution
+      ) do
+    with {:ok, distribution} <- MiningPoolsDistribution.distribution(from, to, interval) do
```
Comment (tspenov) [inline lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: `case` for one clause, `with` for nested

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex [hunk 2]
```diff
@@ -30,6 +31,22 @@ defmodule SanbaseWeb.Graphql.Resolvers.ClickhouseResolver do
...
+
+      {:error, error} ->
+        error_msg = "Can't calculate mining pools distribution."
+        Logger.warn(error_msg <> " Reason: #{inspect(error)}")
+        {:error, error_msg}
+    end
+  end
+
   def mvrv_ratio(_root, args, _resolution) do
     # TODO: Check if interval is a whole day as in token circulation
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: I know see that we have this TODO pushed to master

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex [hunk 3]
```diff
@@ -30,6 +31,22 @@ defmodule SanbaseWeb.Graphql.Resolvers.ClickhouseResolver do
...
+  def mining_pools_distribution(
+        _root,
+        %{from: from, to: to, interval: interval},
+        _resolution
+      ) do
+    case MiningPoolsDistribution.distribution(from, to, interval) do
+      {:ok, distribution} ->
+        {:ok, distribution}
+
+      {:error, error} ->
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: one test that passes through this case

Comment (kalkov) [reply lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: There is [one](https://github.com/santiment/sanbase2/blob/2873cc3d37fed5c9e8cfc6b5af673370cb1587ed/test/sanbase_web/graphql/clickhouse/mining_pools_distribution_test.exs#L75)

**Comment by IvanIvanoff** [comment]
LGTM apart from adding some descriptions. It's not really clear what `miners distribution` mean. What is being distributed among them?

**Pull request merged**